sklearn/ensemble/gradient_boosting.py

```diff
@@ -1367,9 +1381,12 @@ def staged_predict_proba(self, X):
         try:
             for score in self.staged_decision_function(X):
                 yield self.loss_._score_to_proba(score)
-        except AttributeError:
-            raise AttributeError('loss=%r does not support predict_proba' %
-                                 self.loss)
+        except AttributeError as e:
+            if isinstance(e, NotFittedError):
+                raise e
+            else:
+                raise AttributeError('loss=%r does not support predict_proba' %
+                                     self.loss)
```

Comment:
Sorry I am unable to understand that comment... Would you mind elaborating a bit more please?

Earlier `AttributeError` was raised and catched and re-raised with a custom error message... 
But since now we have the possibility of [this](https://github.com/scikit-learn/scikit-learn/pull/4029#discussion-diff-22482923R1383) line raising `NotFittedError` which also gets caught as an `AttributeError`, e... I had added this check for `NotFittedError` to distinguish between those situations...